src/mbgl/style/sources/geojson_source_impl.cpp

```diff
@@ -18,14 +20,30 @@ using namespace mapbox::geojsonvt;
 namespace mbgl {
 namespace style {
 namespace conversion {
+
+struct ToFeatureCollection {
+    mapbox::geojson::feature_collection operator()(const mapbox::geojson::feature_collection value) const {
```

Comment:
The `value` in each visitor here could be passed by `const &` to avoid an unneeded copy. That said, the compiler will likely optimize away the copy so its not certain my suggestion will run any faster.